openssl/hmac.go

```diff
@@ -152,11 +101,11 @@ func (h *opensslHMAC) Sum(in []byte) []byte {
 	// that Sum has no effect on the underlying stream.
 	// In particular it is OK to Sum, then Write more, then Sum again,
 	// and the second Sum acts as if the first didn't happen.
-	h.ctx2 = C.go_openssl_HMAC_CTX_new()
-	if C.go_openssl_HMAC_CTX_copy(h.ctx2, h.ctx) == 0 {
-		panic("openssl: HMAC_CTX_copy_ex failed")
+	ctx2 := C.go_openssl_HMAC_CTX_new()
+	if C.go_openssl_HMAC_CTX_copy(ctx2, h.ctx) == 0 {
+		panic("openssl: HMAC_CTX_copy failed")
 	}
-	C.go_openssl_HMAC_Final(h.ctx2, (*C.uint8_t)(unsafe.Pointer(&h.sum[0])), nil)
-	C.go_openssl_HMAC_CTX_free(h.ctx2)
+	C.go_openssl_HMAC_Final(ctx2, base(h.sum), nil)
+	C.go_openssl_HMAC_CTX_free(ctx2)
```

Comment:
Maybe not something to change in this PR, but should we `defer` the free so that if someone recovers the panic, the context can still get freed and avoid a (I assume) memory leak?

Reply:
Defering the context free cost nothing, so let's do it in this PR 😄